### PR TITLE
Update transactions.md document

### PR DESCRIPTION
**Pull request by akarabashov**

### CODE REVIEW

File: docs/transactions.md [hunk 1]
```diff
@@ -827,13 +827,14 @@ will be in a pending state until sufficient number of approvals is received.
 
 The certificate is immutable. It can only be revoked by either the owner or a quorum of Trustees.
```
Comment (ashcherbakov) [inline docs/transactions.md:829]: What if we rename the commands as follows:
# X509 PKI
## Device Attestation Certificates (DA)
### PROPOSE_ADD_PAA
### APPROVE_ADD_PAA
### REJECT_ADD_PAA
### PROPOSE_REVOKE_PAA
### APPROVE_REVOKE_PAA
### ASSIGN_VID_TO_PAA
### ADD_REVOCATION_DISTRIBUTION_POINT
### UPDATE_REVOCATION_DISTRIBUTION_POINT
### DELETE_REVOCATION_DISTRIBUTION_POINT
### ADD_PAI
### REVOKE_PAI
### REMOVE_PAI
### GET_REVOCATION_DISTRIBUTION_POINT
### GET_REVOCATION_DISTRIBUTION_POINT_BY_SKID
### GET_ALL_REVOCATION_DISTRIBUTION_POINT
### GET_PROPOSED_PAA
### GET_REJECTED_PAA
### GET_PROPOSED_REVOKE_PAA
### GET_ALL_PAA
### GET_ALL_REVOKED_PAA
### GET_ALL_PROPOSED_PAA
### GET_ALL_REJECTED_PAA
### GET_ALL_PROPOSED_REVOKE_PAA
## E2E (NOC/ICA)
### ADD_NOC
### REVOKE_NOC
### ADD_ICA
### GET_NOC_BY_VID
### GET_ICA_BY_VID
### GET_REVOKED_NOC
### GET_ALL_NOC
### GET_ALL_ICA
### GET_ALL_REVOKED_NOC
## Common
### GET_CERT
### GET_REVOKED_CERT
### GET_CERTS_BY_SKID
### GET_CERTS_BY_SUBJECT
### GET_CHILD_CERTS
### GET_ALL_CERTS
### GET_ALL_REVOKED_CERTS

File: docs/transactions.md [hunk 2]
```diff
@@ -1173,16 +1198,6 @@ already present on the ledger.
...
-- Parameters:
-  - cert: `string` - The NOC non-root Certificate, encoded in X.509v3 PEM format. Can be a PEM string or a file path.
-- In State:
-  - `pki/ApprovedCertificates/value/<Subject>/<SubjectKeyID>`
-  - `pki/ApprovedCertificatesBySubject/value/<Subject>`
-  - `pki/ApprovedCertificatesBySubjectKeyID/value/<SubjectKeyID>`
-  - `pki/NOCCertificates/value/<VID>`
-  - `pki/ChildCertificates/value/<Certificate's Subject>/<Certificate's Subject Key ID>`
-- CLI Command:
-  - `dcld tx pki add-noc-x509-cert --certificate=<string-or-path> --from=<account>`
```
Comment (ashcherbakov) [inline docs/transactions.md:1201]: There are no "non root" NOC certificates. NOC == Root. Intermediate must be called ICA. Please fix everywhere in the doc.

**Review by akarabashov — COMMENTED** [review]
I noticed that unit tests for static validation are missing, I think it makes sense to add them

File: docs/transactions.md [hunk 3]
```diff
@@ -827,13 +827,14 @@ will be in a pending state until sufficient number of approvals is received.
...
 The certificate is immutable. It can only be revoked by either the owner or a quorum of Trustees.
 
+- Who can send:
+  - Trustee
 - Parameters:
   - cert: `string` - PEM encoded certificate. The corresponding CLI parameter can contain either a PEM string or a path to a file containing the data.
-  - info: `optional(string)` - information/notes for the proposal
+  - info: `optional(string)` - information/notes for the proposal. Can contain up to 4096 characters.
   - time: `optional(int64)` - proposal time (number of nanoseconds elapsed since January 1, 1970 UTC). CLI uses the current time for that field.
+  - vid: `uint16` -  Vendor ID (positive non-zero). Must be the same as Vendor account's VID. The value must be within the range of 1 to 65535.
```
[the remark this answers is not in the document]
Comment (akarabashov) [reply docs/transactions.md]: Done

File: docs/transactions.md [hunk 4]
```diff
@@ -903,8 +906,6 @@ The certificate is not reject until sufficient number of Trustees reject it.
 Adds an intermediate or leaf X509 certificate signed by a chain of certificates which must be
 already present on the ledger.
 
-The certificate is immutable. It can only be revoked by either the owner or a quorum of Trustees.
-
 - Who can send: Vendor account
   - PAA (Root certificates) are VID-scoped:
```
[the remark this answers is not in the document]
Comment (akarabashov) [reply docs/transactions.md]: Done

File: docs/transactions.md [hunk 5]
```diff
@@ -925,9 +926,10 @@ The certificate is immutable. It can only be revoked by either the owner or a qu
   - no existing certificate with the same `<Certificate's Issuer>:<Certificate's Serial Number>` combination.
   - if certificates with the same `<Certificate's Subject>:<Certificate's Subject Key ID>` combination already exist:
     - the existing certificate must not be NOC certificate
-    - the sender's VID must match the `vid` field of the existing certificates.
+    - the sender's VID must match the VID of the existing certificate's owner.
   - the signature (self-signature) and expiration date are valid.
```
[the remark this answers is not in the document]
Comment (akarabashov) [reply docs/transactions.md]: Removed

File: docs/transactions.md (hunk 2 above)
Comment (akarabashov) [reply docs/transactions.md:1121]: This will be implemented in a separate pull request.

File: docs/transactions.md (hunk 1 above)
Comment (akarabashov) [reply docs/transactions.md:841]: Renamed

File: docs/transactions.md [hunk 6]
```diff
@@ -828,26 +828,29 @@ Should be sent to trusted nodes only.
...
 
-### PROPOSE_ADD_X509_ROOT_CERT
+### Device Attestation Certificates (DA)
+
+#### PROPOSE_ADD_PAA
 
 **Status: Implemented**
 
-Proposes a new self-signed root certificate.
+Proposes a new PAA certificate.
```
Comment (ashcherbakov) [inline docs/transactions.md:837]: Please clarify that a PAA is a root self-signed certificate.

File: docs/transactions.md [hunk 7]
```diff
@@ -1149,305 +1083,213 @@ Deletes a PKI Revocation distribution endpoint (such as RFC5280 Certificate Revo
...
+    - the `vid` in the subjects of both certificates must be equal to the sender Vendor account's VID.
+  - if the parent root certificate is not VID scoped but has an associated VID:
+    - the provided certificate can be either VID scoped or non-VID scoped.
+    - if the provided certificate is VID scoped, the `vid` in the subject of the certificate must be equal to the VID associated with the root certificate and to the sender Vendor account's VID.
+  - if the parent root certificate is non-VID scoped and does not have an associated VID, an error will occur.
 
-### REVOKE_NOC_X509_ROOT_CERT
+> **_Note:_**  Multiple certificates can refer to the same `<Certificate's Subject>:<Certificate's Subject Key ID>` combination.
+
+#### REVOKE_PAI
```
Comment (ashcherbakov) [inline docs/transactions.md:1124]: Please add a clarification how it's different from REMOVE_PAI

File: docs/transactions.md [hunk 8]
```diff
@@ -1149,305 +1083,213 @@ Deletes a PKI Revocation distribution endpoint (such as RFC5280 Certificate Revo
...
 - CLI command:
-  - `dcld query pki x509-cert --subject-key-id=<hex string>`
-- REST API:
-  - GET `/dcl/pki/certificates?subjectKeyId={subjectKeyId}`
+  - `dcld tx pki revoke-x509-cert --subject=<base64 string> --subject-key-id=<hex string> --from=<account>`
+- Validation:
+  - a PAI Certificate with the provided `subject` and `subject_key_id` must exist in the ledger.
 
-### GET_ALL_SUBJECT_X509_CERTS
+#### REMOVE_PAI
```
Comment (ashcherbakov) [inline docs/transactions.md:1153]: Please add a clarification how it's different from REVOKE_PAI

File: docs/transactions.md [hunk 9]
```diff
@@ -1495,55 +1337,124 @@ Should be sent to trusted nodes only.
...
+    - `Issuer` == `Subject`
+    - `Authority Key Identifier` == `Subject Key Identifier`
+  - no existing certificate with the same `<Certificate's Issuer>:<Certificate's Serial Number>` combination.
+  - if certificates with the same `<Certificate's Subject>:<Certificate's Subject Key ID>` combination already exist:
+    - the existing certificate must be NOC root certificate
+    - the sender's VID must match the `vid` field of the existing certificates.
+  - the signature (self-signature) and expiration date must be valid.
 
-### GET_PKI_REVOCATION_DISTRIBUTION_POINTS_BY_SUBJECT_KEY_ID
+#### REVOKE_NOC_ROOT
```
Comment (ashcherbakov) [inline docs/transactions.md:1371]: Please clarify that revoke here means soft-delete (revoke certificates can be queried)

File: docs/transactions.md [hunk 10]
```diff
@@ -1495,55 +1337,124 @@ Should be sent to trusted nodes only.
...
+- In State:
+  - `pki/ApprovedCertificates/value/<Subject>/<SubjectKeyID>`
+  - `pki/ApprovedCertificatesBySubject/value/<Subject>`
+  - `pki/ApprovedCertificatesBySubjectKeyID/value/<SubjectKeyID>`
+  - `pki/NocIcaCertificates/value/<VID>`
+  - `pki/ChildCertificates/value/<Certificate's Subject>/<Certificate's Subject Key ID>`
+- CLI Command:
+  - `dcld tx pki add-noc-x509-cert --certificate=<string-or-path> --from=<account>`
+
+#### REVOKE_NOC_ICA
```
Comment (ashcherbakov) [inline docs/transactions.md:1429]: Please clarify that revoke here means soft-delete (revoke certificates can be queried)

File: docs/transactions.md [hunk 11]
```diff
@@ -1607,19 +1532,123 @@ Gets all revoked NOC root certificates.
 - REST API:
   - GET `/dcl/pki/revoked-noc-root-certificates`
 
-### GET_REVOKED_NOC_X509_ROOT_CERT
+### Common
+
+#### GET_CERT
```
Comment (ashcherbakov) [inline docs/transactions.md:1537]: Please clarify that it works for all types of certificates (PAA, PAI, NOC_ROOT, NOC_ICA).

File: docs/transactions.md [hunk 12]
```diff
@@ -1607,19 +1532,123 @@ Gets all revoked NOC root certificates.
...
+  - subject: `string`  - certificates's `Subject` is base64 encoded subject DER sequence bytes
+  - subject_key_id: `string`  - certificates's `Subject Key Id` in hex string format, e.g: `5A:88:0E:6C:36:53:D0:7F:B0:89:71:A3:F4:73:79:09:30:E6:2B:DB`
 - CLI command:
-  - `dcld query pki revoked-noc-x509-root-cert --subject=<base64 string> --subject-key-id=<hex string>`
+  - `dcld query pki x509-cert --subject=<base64 string> --subject-key-id=<hex string>`
 - REST API:
-  - GET `/dcl/pki/revoked-noc-root-certificates/{subject}/{subject_key_id}`
+  - GET `/dcl/pki/certificates/{subject}/{subject_key_id}`
+
+#### GET_REVOKED_CERT
```
Comment (ashcherbakov) [inline docs/transactions.md:1553]: Please clarify that it works for all types of certificates (PAA, PAI, NOC_ROOT, NOC_ICA).

File: docs/transactions.md [hunk 13]
```diff
@@ -1607,19 +1532,123 @@ Gets all revoked NOC root certificates.
...
+
+- Parameters:
+  - subject: `string`  - certificates's `Subject` is base64 encoded subject DER sequence bytes
+  - subject_key_id: `string`  - certificates's `Subject Key Id` in hex string format, e.g: `5A:88:0E:6C:36:53:D0:7F:B0:89:71:A3:F4:73:79:09:30:E6:2B:DB`
+- CLI command:
+  - `dcld query pki revoked-x509-cert --subject=<base64 string> --subject-key-id=<hex string>`
+- REST API:
+  - GET `/dcl/pki/revoked-certificates/{subject}/{subject_key_id}`
+
+#### GET_CERTS_BY_SKID
```
Comment (ashcherbakov) [inline docs/transactions.md:1570]: Please clarify that it works for all types of certificates (PAA, PAI, NOC_ROOT, NOC_ICA).

File: docs/transactions.md [hunk 14]
```diff
@@ -1607,19 +1532,123 @@ Gets all revoked NOC root certificates.
...
+Use `GET_ALL_REVOKED_CERTS` to get a list of all revoked certificates.
+
+- Parameters:
+  - subject_key_id: `string`  - certificates's `Subject Key Id` in hex string format, e.g: `5A:88:0E:6C:36:53:D0:7F:B0:89:71:A3:F4:73:79:09:30:E6:2B:DB`
+- CLI command:
+  - `dcld query pki x509-cert --subject-key-id=<hex string>`
+- REST API:
+  - GET `/dcl/pki/certificates?subjectKeyId={subjectKeyId}`
+
+#### GET_CERTS_BY_SUBJECT
```
Comment (ashcherbakov) [inline docs/transactions.md:1586]: Please clarify that it works for all types of certificates (PAA, PAI, NOC_ROOT, NOC_ICA).

File: docs/transactions.md [hunk 15]
```diff
@@ -1607,19 +1532,123 @@ Gets all revoked NOC root certificates.
...
+Use [GET_ALL_REVOKED_CERTS](#get_all_revoked_certs) to get a list of all revoked certificates.
+
+- Parameters:
+  - subject: `string`  - certificates's `Subject` is base64 encoded subject DER sequence bytes
+- CLI command:
+  - `dcld query pki all-subject-x509-certs --subject=<base64 string>`
+- REST API:
+  - GET `/dcl/pki/certificates/{subject}`
+
+#### GET_CHILD_CERTS
```
Comment (ashcherbakov) [inline docs/transactions.md:1602]: Please clarify that it works for all types of certificates (PAA, PAI, NOC_ROOT, NOC_ICA).

File: docs/transactions.md [hunk 16]
```diff
@@ -1607,19 +1532,123 @@ Gets all revoked NOC root certificates.
...
+
+- Parameters:
+  - subject: `string`  - certificates's `Subject` is base64 encoded subject DER sequence bytes
+  - subject_key_id: `string`  - certificates's `Subject Key Id` in hex string format, e.g: `5A:88:0E:6C:36:53:D0:7F:B0:89:71:A3:F4:73:79:09:30:E6:2B:DB`
+- CLI command:
+  - `dcld query pki all-child-x509-certs (--subject=<base64 string> --subject-key-id=<hex string>`
+- REST API:
+  - GET `/dcl/pki/child-certificates/{subject}/{subject_key_id}`
+
+#### GET_ALL_CERTS
```
Comment (ashcherbakov) [inline docs/transactions.md:1617]: Please clarify that it returns all certificates (PAA, PAI, NOC_ROOT, NOC_ICA).

File: docs/transactions.md [hunk 17]
```diff
@@ -1607,19 +1532,123 @@ Gets all revoked NOC root certificates.
...
+Should be sent to trusted nodes only.
+
+- Parameters:
+  - Common pagination parameters (see [pagination-params](#common-pagination-parameters))
+- CLI command:
+  - `dcld query pki all-x509-certs`
+- REST API:
+  - GET `/dcl/pki/certificates`
+
+#### GET_ALL_REVOKED_CERTS
```
Comment (ashcherbakov) [inline docs/transactions.md:1635]: Please clarify that it returns all revoked certificates (PAA, PAI, NOC_ROOT, NOC_ICA).

File: docs/transactions.md (hunk 7 above)
Comment (akarabashov) [reply docs/transactions.md:1127]: Done

File: docs/transactions.md (hunk 9 above)
Comment (akarabashov) [reply docs/transactions.md:1377]: Done

File: docs/transactions.md (hunk 10 above)
Comment (akarabashov) [reply docs/transactions.md:1438]: Done

File: docs/transactions.md (hunk 11 above)
Comment (akarabashov) [reply docs/transactions.md:1565]: Donw

File: docs/transactions.md (hunk 12 above)
Comment (akarabashov) [reply docs/transactions.md:1581]: Done

File: docs/transactions.md (hunk 13 above)
Comment (akarabashov) [reply docs/transactions.md:1598]: Done

File: docs/transactions.md (hunk 14 above)
Comment (akarabashov) [reply docs/transactions.md:1614]: Done

File: docs/transactions.md (hunk 15 above)
Comment (akarabashov) [reply docs/transactions.md:1630]: Done

**Review by ashcherbakov — APPROVED** [review]

**Pull request merged**